### PR TITLE
PI-12 Fix end-to-end tests for VPD

### PR DESCRIPTION
**Pull request by marcus-bcl**

### CODE REVIEW

File: src/e2e/kotlin/uk/gov/justice/digital/hmpps/deliusapi/EndToEndTest.kt [hunk 1]
```diff
@@ -85,31 +86,33 @@ abstract class EndToEndTest {
   }
 
   private fun getToken(): String {
-    if (CACHED_TOKEN?.expired() == false) {
-      return CACHED_TOKEN!!.accessToken
+    if (CACHED_TOKEN?.expired() != false) {
```
Comment (mxco86) [inline src/e2e/kotlin/uk/gov/justice/digital/hmpps/deliusapi/EndToEndTest.kt:89]: Can you remove the condition completely here?

File: doc/testing.md [hunk 2]
```diff
@@ -59,18 +59,36 @@ database for test setup & teardown. Environment configuration is handled by
...
 
-To run the end-to-end tests against the local docker oracle database, use the
-`local-oracle` spring profile.
-This is configured by default to use the `delius-api-client` client from the auth service.
+To run the end-to-end tests against local Oracle database and Elasticsearch 
+containers, use the `local-oracle` and `local-elasticsearch` Spring profiles.
 
 ```bash
-SPRING_PROFILES_ACTIVE=local-oracle ./gradlew e2e
+SPRING_PROFILES_ACTIVE=local-oracle,local-elasticsearch ./gradlew e2e
```
Comment (mxco86) [inline doc/testing.md:68]: These profiles being slightly different from the application is a bit confusing - might be worth picking one or the other

**Review by mxco86 — APPROVED** [review]
LGTM - couple of small things

File: src/e2e/kotlin/uk/gov/justice/digital/hmpps/deliusapi/EndToEndTest.kt (hunk 1 above)
Comment (marcus-bcl) [reply src/e2e/kotlin/uk/gov/justice/digital/hmpps/deliusapi/EndToEndTest.kt:89]: I don't think so, it's effectively saying - if the CACHED_TOKEN is `null`, or CACHED_TOKEN.expired() is `true`, then fetch the token.  It is confusing though, maybe being more explicit is better e.g.
```suggestion
    if (CACHED_TOKEN == null || CACHED_TOKEN.expired()) {
```

File: doc/testing.md (hunk 2 above)
Comment (marcus-bcl) [reply doc/testing.md:68]: Yeah agreed, it's not super obvious.  Maybe we just have `application-local.yml` for running against local containers, and `application-test.yml` for running against AWS Test?

File: src/e2e/kotlin/uk/gov/justice/digital/hmpps/deliusapi/EndToEndTest.kt (hunk 1 above)
Comment (mxco86) [reply src/e2e/kotlin/uk/gov/justice/digital/hmpps/deliusapi/EndToEndTest.kt:89]: OK yes - that is clearer

File: doc/testing.md (hunk 2 above)
Comment (mxco86) [reply doc/testing.md:68]: Yep makes sense - we can do another ticket for rationalising these

**Review by mxco86 — APPROVED** [review]

**Pull request merged**